packages/clay-autocomplete/stories/index.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayAutocomplete from '../src';
+import ClayDropDown from '@clayui/drop-down';
+import fuzzy from 'fuzzy';
+import React, {useState} from 'react';
+import {FetchPolicy, NetworkStatus} from '@clayui/data-provider/src/types';
+import {storiesOf} from '@storybook/react';
+import {useDebounce} from '@clayui/shared';
+import {useResource} from '@clayui/data-provider';
+
+import 'clay-css/lib/css/atlas.css';
+
+const LoadingWithDebounce = ({
+	loading,
+	networkStatus,
+	render,
+}: {
+	networkStatus?: NetworkStatus;
+	loading: boolean;
+	render: any;
+}) => {
+	const debouncedLoadingChange = useDebounce(loading, 500);
+
+	if (networkStatus === 1 || debouncedLoadingChange) {
+		return (
+			<ClayDropDown.Item className="disabled">
+				{'Loading...'}
+			</ClayDropDown.Item>
+		);
+	}
+
+	return render;
+};
+
+const optionsFuzzy = {post: '</strong>', pre: '<strong>'};
+
+interface IProps {
+	name: string;
+	value: string;
+}
+
+const Item: React.FunctionComponent<IProps> = ({name, value}) => {
+	const match = fuzzy.match(value, name, optionsFuzzy);
+
+	if (match) {
+		return <div dangerouslySetInnerHTML={{__html: match.rendered}} />;
+	}
+
+	return <>{name}</>;
+};
+
+const ClayAutocompleteWithState = () => {
+	const [value, setValue] = useState('');
+	const [networkStatus, setNetworkStatus] = useState<NetworkStatus>(
+		NetworkStatus.Unused
+	);
+	const {resource} = useResource({
+		fetchPolicy: FetchPolicy.CacheFirst,
+		link: 'https://rickandmortyapi.com/api/character/',
+		onNetworkStatusChange: setNetworkStatus,
+		variables: {name: value},
+	});
+
+	const initialLoading = networkStatus === 1;
+	const loading = networkStatus < 4;
+	const error = networkStatus === 5;
+
+	return (
+		<ClayAutocomplete>
+			<ClayAutocomplete.Input
+				onChange={event => setValue(event.target.value)}
+				value={value}
+			/>
+			<ClayAutocomplete.DropDown
+				active={(!!resource && !!value) || initialLoading}
+			>
+				<ClayDropDown.ItemList>
+					<LoadingWithDebounce
```

Comment:
I've been wondering if this could turn out to be a high-level or low-level component, part of a specification to give feedback to the user, will probably be heavily used. What do you think @bryceosterhaus?

Reply:
Not sure if it should be its own component or if we just create a shared hook. I know once the suspense API is fully released that will help with the integration point of using loading indicators. I'd probably be fine if it was its own component though, we can always try it out and see how it works.

Reply:
Hm I'm not sure if Suspense would help with this, this rule is very specific, the loading screen in DropDown should only appear when the data is passing 500ms. But I think we can run out of it for now and document this use case in next.clayui.com.